File: src/main/java/ru/openbank/meetup/demo/java/slide5/WhenUseCase2.java
package ru.openbank.meetup.demo.java.slide5;

import org.springframework.util.StringUtils;

public class WhenUseCase2 {

    String formatPhoneNumber(String phone) {
        if (!StringUtils.isEmpty(phone) && phone.length() > 10) {
            return phone.substring(phone.length() - 10);
        }
        return phone;
    }
}

File: src/main/kotlin/ru/openbank/meetup/demo/kotlin/slide5/WhenUseCase2.kt
package ru.openbank.meetup.demo.kotlin.slide5

fun formatPhoneNumber(phone: String?): String? {
    return when (phone.isTooLong()) {
        true -> phone?.substring(phone.length - MAX_PHONE_LENGTH)
        false -> phone
    }
}

private fun String?.isTooLong(): Boolean {
    return !this.isNullOrEmpty() && this.length > MAX_PHONE_LENGTH
}

private const val MAX_PHONE_LENGTH = 10
